theano/compile/debugmode.py

```diff
@@ -2293,7 +2298,7 @@ def __init__(self, inputs, outputs, optimizer, mode,
                 for output, spec in izip(fgraph.outputs,
                                          outputs + additional_outputs)
                 if not spec.borrow]
-        if no_borrow:
+        if no_borrow and not always_borrow:
```

Comment:
I think we should remove the always_borrow parameter everywhere and use the existing system: wrapping an output variable with an Out object:
```
v=theano.tensor.vector()
theano.Out(v, borrow=True)
```